wai-extra/Network/Wai/Parse.hs

```diff
@@ -115,6 +118,35 @@ tempFileBackEndOpts getTmpDir pattern internalState _ _ popper = do
             loop
     release key
     return fp
+    where
+        it = do
+            tempDir <- getTmpDir
+            openBinaryTempFile tempDir pattrn
+
+-- | A data structure that describes the behavior of
+-- the parseRequestBodyEx function.
+data ParseRequestBodyOptions = ParseRequestBodyOptions
+    { -- | The maximum length of a filename
+      prboKeyLength             :: Int
+    , -- | The maximum number of files.
+      prboMaxNumFiles           :: Int
+    , -- | The maximum filesize per file.
+      prboMaxFileSize           :: Maybe Int64
+    , -- | The maximum total filesize.
+      prboMaxFilesSize          :: Maybe Int64
+    , -- | The maximum size of the sum of all parameters values
+      prboMaxParmsValueSize     :: Int
+    , -- | The maximum header lines per entry
+      prboMaxHeaderLines        :: Int }
+
+instance Default ParseRequestBodyOptions where
```

Comment:
Most of the community has been moving away from the `Default` typeclass, how about just having a `defaultParseRequestBodyOptions` value?